server/src/internalClusterTest/java/org/opensearch/indices/DateMathIndexExpressionsIntegrationIT.java

```diff
@@ -77,7 +77,7 @@ public void testIndexNameDateMathExpressions() {
         refresh();
 
         SearchResponse searchResponse = client().prepareSearch(dateMathExp1, dateMathExp2, dateMathExp3).get();
-        assertHitCount(searchResponse, 3);
+        assertBusy(() -> assertHitCount(searchResponse, 3));
```

Comment:
I think assertBusy() will not fix the issue here because the value of searchResponse remains same every time. Once it fails you need to make another SearchResponse API call to get new searchResponse value. May be something like this would help:
```
assertBusy(() -> {
SearchResponse searchFinalResponse = client().prepareSearch(dateMathExp1, dateMathExp2, dateMathExp3).get();
assertHitCount(searchFinalResponse, 3)
});
```

Reply:
Good catch @Rishikesh1159